From 3a00c741bf0057e4b73e92107b701433a6be026f Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:33:12 +0200
Subject: [PATCH 30/65] acerhk: disable on EFI hardware

This driver directly accesses memory values and all devices supported by this
driver predate UEFI by several years, don't risk damaging broken UEFI systems.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -56,6 +56,7 @@
 #include <linux/proc_fs.h>
 #include <linux/uaccess.h>
 #include <linux/delay.h>
+#include <linux/efi.h>
 
 #include "acerhk.h"
 
@@ -1806,6 +1807,9 @@ static int __init acerhk_init(void)
 {
 	int error;
 
+	if (efi_enabled(EFI_BOOT))
+		return -ENODEV;
+
 	error = platform_driver_register(&acerhk_driver);
 	if (error)
 		return error;
